schema/messaging.proto

```diff
@@ -0,0 +1,463 @@
+  // "message.update_time" fields will be ignored by the server in this request.
+  Message message = 2;
+}
+
+// The request message for the google.showcase.v1alpha3.Messaging\GetMessage
+// method.
+message GetMessageRequest {
+  // The URI of the requested message.
+  string name = 1 [
+    (google.api.resource_reference) = "google.showcase.v1alpha3.Message",
+    (google.api.field_behavior) = REQUIRED
+  ];
+}
+
+// The request message for the google.showcase.v1alpha3.Messaging\UpdateMessage
+// method.
+message UpdateMessageRequest {
+  // The message to be updated.
+  Message message = 1;
+
+  // The field mask to determine wich fields are to be updated. If empty, the
+  // server will assume all fields are to be updated.
+  google.protobuf.FieldMask update_mask = 2;
+}
+
+// The request message for the google.showcase.v1alpha3.Messaging\DeleteMessage
+// method.
+message DeleteMessageRequest {
+  // The URI of the requested message.
+  string name = 1 [
+    (google.api.resource_reference) = "google.showcase.v1alpha3.Message",
+    (google.api.field_behavior) = REQUIRED
+  ];
+}
+
+// The request message for the google.showcase.v1alpha3.Messaging\ListMessages
+// method.
+message ListMessagesRequest {
+  // The URI of the requested room or profile whos messages are to be listed.
+  string parent = 1 [
+    (google.api.resource_reference) = "google.showcase.v1alpha3.MessageParent",
+    (google.api.field_behavior) = REQUIRED
+  ];
+
+
+  // The maximum number of rooms to be returned. Server may return fewer rooms
+  // than requested. If unspecified, server will pick an appropriate default.
+  int32 page_size = 2;
+
+  // Typically, this is the value of
+  // google.showcase.v1alpha3.ListMessagesResponse.next_page_token
+  // returned from the previous call to
+  // `google.showcase.v1alpha3.Messaging\ListMessages` method.
+  string page_token = 3;
+}
+
+// The response message for the google.showcase.v1alpha3.Messaging\ListMessages
+// method.
+message ListMessagesResponse {
+  // The list of messages.
+  repeated Message messages = 1;
+
+  // A token to retrieve next page of results.
+  // Pass this value in ListMessagesRequest.page_token field in the subsequent
+  // call to `google.showcase.v1alpha3.Message\ListMessages` method to retrieve
+  // the next page of results.
+  string next_page_token = 2;
+}
+
+// The request message for the google.showcase.v1alpha3.Messaging\SearchMessages
+// method.
+message SearchMessagesRequest {
+  // The query used to search for messages containing to words of this string.
+  // Only posts that contain an exact match of a queried word will be returned.
+  string query = 1 [(google.api.field_behavior) = REQUIRED];
+}
+
+// The operation metadata message for the
+// google.showcase.v1alpha3.Messaging\SearchMessages method.
+message SearchMessagesMetadata {
+  // This signals to the client to when to next poll for response.
+  google.rpc.RetryInfo retry_info = 1;
```

Comment:
OT: I don't see this mentioned in API style guide. Should our generators make use of this?

Reply:
+1. I don't think this message has wide adoption amongst google APIs but building a GAPIC feature on top of this would be great. 

The only API I can think of that is returning this message is Spanner by returning this as part of their response headers for certain methods. /shrug if it is documented well though.